### PR TITLE
Support run-bench targeting local instance of rebuilder.

### PR DESCRIPTION
**Pull request by wbxyz**
Change the run-bench to detecting a url that looks like a cloud run url (the same way run-one does). Based on this switch, run-bench will only conditionally upgrade to https or include an oauth token.

When the `--build-local` bool flag is provided, run-bench will skip the runid creation process, and it will also skip the warmup checks.

To help with running the local builder, the user agent is now optional.

### CODE REVIEW

File: tools/ctl/ctl.go [hunk 1]
```diff
@@ -523,6 +534,7 @@ var (
 func init() {
 	runBenchmark.Flags().AddGoFlag(flag.Lookup("api"))
 	runBenchmark.Flags().AddGoFlag(flag.Lookup("max-concurrency"))
+	runBenchmark.Flags().AddGoFlag(flag.Lookup("build-local"))
```
Comment (msuozzo) [inline tools/ctl/ctl.go:537]: what about just `--local`

File: tools/ctl/ctl.go [hunk 2]
```diff
@@ -269,7 +275,11 @@ var runBenchmark = &cobra.Command{
 			log.Fatal(err)
 		}
 		var run string
-		{
+		if *buildLocal {
+			// The build-local service does not support creating a new run-id.
+			// If we're talking to build-local directly, then we skip run-id generation.
+			run = "build-local-run"
```
Comment (msuozzo) [inline tools/ctl/ctl.go:281]: might still want to generate a timestamp-like ID

File: tools/ctl/ctl.go [hunk 3]
```diff
@@ -252,11 +252,17 @@ var runBenchmark = &cobra.Command{
 		if err != nil {
 			log.Fatal(errors.Wrap(err, "parsing API endpoint"))
 		}
-		apiURL.Scheme = "https"
 		ctx := cmd.Context()
-		idclient, err := oauth.AuthorizedUserIDClient(ctx)
-		if err != nil {
-			log.Fatal(errors.Wrap(err, "creating authorized HTTP client"))
+		var idclient *http.Client
+		if strings.Contains(apiURL.Host, "run.app") {
```
Comment (msuozzo) [inline tools/ctl/ctl.go:257]: HasSuffix? and that should probably be `.run.app`

**Review by msuozzo — APPROVED** [review]

**Review by msuozzo — CHANGES_REQUESTED** [review]
Can you also format the PR titles in the imperative? -> "Support local run-bench"

File: tools/ctl/ctl.go (hunk 3 above)
Comment (wbxyz) [reply tools/ctl/ctl.go:257]: Resolved in https://github.com/google/oss-rebuild/pull/35

File: tools/ctl/ctl.go [hunk 4]
```diff
@@ -363,7 +373,7 @@ var runBenchmark = &cobra.Command{
 						}
 						if resp.StatusCode != 200 {
 							totalErrors++
-							aggErrors = append(aggErrors, errors.Wrap(errors.New(resp.Status), "request").Error())
+							aggErrors = append(aggErrors, errors.Wrapf(errors.New(resp.Status), "request: %s", req.URL.String()).Error())
```
Comment (msuozzo) [inline tools/ctl/ctl.go:376]: nit: "requesting %s"

**Review by msuozzo — APPROVED** [review]

**Pull request merged**